### PR TITLE
Use ClimaCore grid

### PR DESCRIPTION
**Pull request by charleskawczynski**
This PR switches over to using a ClimaCore.jl grid and changes some variable names
 - `dz` -> `Δz`
 - `z_half` -> `zc`
 - `z` -> `zf`

I think that this should all work with [ClimaCore.jl's 173](https://github.com/CliMA/ClimaCore.jl/pull/173). We'll need to create a new branch to update to, or just point to a new SHA, maybe that's easier than creating new branches.

Since this PR changes `z_half` / `z` to `zc` and `zf`, we'll need a follow-up PR to change the names for the main branch.

### CODE REVIEW

File: src/Grid.jl [hunk 1]
```diff
@@ -1,62 +1,58 @@
...
-    dz::FT
-    dzi::FT
+    Δz::FT
+    Δzi::FT
     nz::Int
-    nzg::Int
-    z_half::A1
-    z::A1
+    zc::SC
+    zf::SF
```
Comment (haakon-e) [inline src/Grid.jl:10]: Are these in practice regular arrays, or some CC construct?

File: src/Grid.jl [hunk 2]
```diff
@@ -1,62 +1,58 @@
...
 is_surface_center(grid::Grid, k::Int) = k == kc_surface(grid)
 is_toa_center(grid::Grid, k::Int) = k == kc_top_of_atmos(grid)
 is_surface_face(grid::Grid, k::Int) = k == kf_surface(grid)
 is_toa_face(grid::Grid, k::Int) = k == kf_top_of_atmos(grid)
 
-zc_surface(grid::Grid) = grid.z_half[kc_surface(grid)]
-zf_surface(grid::Grid) = grid.z[kf_surface(grid)]
-zc_toa(grid::Grid) = grid.z_half[kc_top_of_atmos(grid)]
-zf_toa(grid::Grid) = grid.z[kf_top_of_atmos(grid)]
+zc_surface(grid::Grid) = vec(grid.zc)[kc_surface(grid)]
```
Comment (haakon-e) [inline src/Grid.jl:49]: Is the `vec` function related to the lack of support for slicing? (https://github.com/CliMA/ClimaCore.jl/pull/173#pullrequestreview-752224612)

File: src/Turbulence_PrognosticTKE.jl [hunk 3]
```diff
@@ -1794,7 +1792,8 @@ function update_inversion(self::EDMF_PrognosticTKE, GMV::GridMeanVariables, opti
             ∇θ_liq = c∇_downwind(∇θ_liq_cut, grid, k; bottom = FreeBoundary(), top = SetGradient(0))
             if ∇θ_liq > ∇θ_liq_max
                 ∇θ_liq_max = ∇θ_liq
-                self.zi = grid.z[k]
+                # TODO: bugfix: we should be using grid.zc[k]
```
Comment (haakon-e) [inline src/Turbulence_PrognosticTKE.jl:1795]: do we need an issue for this?

File: docs/src/ReferenceStates.md [hunk 4]
```diff
@@ -21,7 +21,7 @@ function export_ref_profile(case_name::String)
     case = Cases.CasesFactory(namelist, grid, ref_state)
     Cases.initialize_reference(case, grid, ref_state, Stats)
     NCDatasets.Dataset(joinpath(Stats.path_plus_file), "r") do ds
-        zc = ds.group["profiles"]["z_half"][:]
+        zc = ds.group["profiles"]["zc"][:]
         zf = ds.group["profiles"]["z"][:]
```
Comment (haakon-e) [inline docs/src/ReferenceStates.md:25]: should this be `"zf"` instead of `"z"`?

**Review by haakon-e — APPROVED** [review]
Overall, LGTM

File: src/Grid.jl (hunk 1 above)
Comment (charleskawczynski) [reply src/Grid.jl:10]: They’re types— finite difference spaces, I think

File: src/Grid.jl (hunk 2 above)
Comment (charleskawczynski) [reply src/Grid.jl:49]: Yeah, I forgot to revert that, thanks!

File: src/Turbulence_PrognosticTKE.jl (hunk 3 above)
Comment (charleskawczynski) [reply src/Turbulence_PrognosticTKE.jl:1795]: Yeah, opened #256.

File: docs/src/ReferenceStates.md (hunk 4 above)
Comment (charleskawczynski) [reply docs/src/ReferenceStates.md:25]: Yes! Good catch!

File: src/NetCDFIO.jl [hunk 5]
```diff
@@ -78,23 +78,23 @@ mutable struct NetCDFIO_Stats
 
         NC.Dataset(path_plus_file, "c") do root_grp
 
-            zf = Gr.z
-            zc = Gr.z_half
+            zf = vec(Gr.zf)
+            zc = vec(Gr.zc)
```
Comment (haakon-e) [inline src/NetCDFIO.jl:82]: did you mean to remove `vec` here too?

File: src/ReferenceState.jl [hunk 6]
```diff
@@ -76,8 +76,8 @@ function initialize(self::ReferenceState, grid::Grid, Stats::NetCDFIO_Stats)
     prob = ODEProblem(rhs, logp, z_span)
     sol = solve(prob, Tsit5(), reltol = 1e-12, abstol = 1e-12)
 
-    p .= sol.(grid.z)
-    p_half .= sol.(grid.z_half)
+    p .= sol.(vec(grid.zf))
+    p_half .= sol.(vec(grid.zc))
```
Comment (haakon-e) [inline src/ReferenceState.jl:80]: and here

File: src/NetCDFIO.jl (hunk 5 above)
Comment (charleskawczynski) [reply src/NetCDFIO.jl:82]: No, I imagine that NCDatasets needs vectors here

File: src/ReferenceState.jl (hunk 6 above)
Comment (charleskawczynski) [reply src/ReferenceState.jl:80]: No, I think that DiffEq’s solution type wouldn’t work without calling `vec` on the field.

**Comment by charleskawczynski** [comment]
@haakon-e, @yairchn, one thing that I'm not crazy about with this PR is that we have to leave `z_half` in a few places, where we read from NC files. Thoughts?

**Comment by haakon-e** [comment]
I don't have a full overview of all these places -- can we update any .nc files if they're static? or would we have to update pycles too?

**Comment by charleskawczynski** [comment]
> I don't have a full overview of all these places -- can we update any .nc files if they're static? or would we have to update pycles too?

Technically we only need to update the NC files, which would change our Artifact.toml (currently [git ignored](https://github.com/CliMA/TurbulenceConvection.jl/blob/main/.gitignore#L34)).

There are some consequences of changing the NC file links, though. After changing it, we no longer can checkout older branches and expect them to run. We could simply add a new set of NC files to the cloud with this name change, but we obviously don't want to do that many times.

I'm almost in favor of leaving legacy names around for some time, idk. Going to sleep on it.

**Comment by charleskawczynski** [comment]
> I don't have a full overview of all these places -- can we update any .nc files if they're static? or would we have to update pycles too?

It’s only for strings in accessing NC files.

I think it’s fine—our data structures will still be consistent, and I think we’ll always be interfacing with data/simulations that have other naming conventions/formulations

**Pull request merged**